### DISCUSSION TITLE
Segmentation of the the transport layer

### DISCUSSION THREAD

**Discussion by HarlekinSimplex** [markqvist/Reticulum#23] [Ideas]
Having a Radio Meshnet that is shared by any client that is within reach is a good thing and that works out of the box already.
Actually I would like to limit and manage the amount of annoucement and message traffic that may occur as soon as we have possibly thousands of clients wihtin range of a radio station or a network of radio stations serving lots of local clusters of LoRa Clients.
The idea I came off with would be to divide the radio frequency used into separated networks.
So having one RNS Station transmitting on frequencey 145.850 within Network#1 does not interfere with another RNS Station transmitting on 145.850 as well but its transmissions are bound to network#2. Announcements and messages in those networks shall only be 'heard' by the mesh network they are in. Cross Network Traffic, if required would than be possible only by implementing a bridge application on top of the segmentet transport layer.
To use several frequencies to separate the readio meshes would required multipe TRXs and Antenna Arrangements beside allowance to use several frequencies. So going that way would more or less beeing no option. Neither with HAM and LoRa nor with any other radio band that we are aiming to use.

Question:
Is such a segmentation possible already on RNS radio interfaces (KISS, KISS X25 or LoRa) ?
If not, is such a feature already on the RoadMap?
If yes, when would it be possibly available?

Cheers
Stephan

**Comment by markqvist** [comment]
Yes, this is completely necessary, and already on the roadmap. In that regard I really should get the current feature and implementation roadmap posted somewhere so it is easier to track what features are currently planned and being worked on.

Either way, this is in the pipeline, and it will probably be included in one of the next releases. I have been holding off on the implementation for a while since I was not sure whether it would be the best decision to delegate this kind of functionality to the link layer devices, which is the established wisdom. There is a lot of points going for the "traditional" solution, but also a pretty strong argument for letting Reticulum itself take care of this. With this functionality in Reticulum itself, it would be possible to implement access control and network segmentation on *any* interface type that Reticulum supports, regardless of the link-layer supporting it natively. That would be a very valuable feature.

**Comment by markqvist** [comment]
Also, @HarlekinSimplex, if you have input or thoughts on the implementation methodology of this feature, I would be interested to hear them.

**Comment by markqvist** [comment]
This is now implemented by 5d90ea565a8d3f640285d9b490d404d539eab699. Will be included in the upcoming 0.3.5 release.
